src/lib/utils/slug.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * If a slug includes a page number (`page-{n}`) in the last segment,
+ * it is a generated path for paging, and Drupal will not know about it.
+ * The page number needs to be extracted and the page segment needs to
+ * be removed from the slug to generate the path.
+ *
+ * Ex:
+ * input: [
+ *  'minneapolis-health-care',
+ *  'stories',
+ *  'page-2'
+ * ]
+ * output: [
+ *  'minneapolis-health-care/stories',
+ *  2
+ * ]
+ *
+ *
+ * In all other cases, the path is formed simply by using all segments
+ * from the slug, and the page number is returned as undefined.
+ *
+ * Ex:
+ * input: [
+ *  'minneapolis-health-care'
+ *  'stories',
+ *  'some-story-title'
+ * ]
+ * output: [
+ *  'minneapolis-health-care/stories/some-story-title`,
+ *  undefined
+ * ]
+ */
+export function getPathAndPageNumberFromSlug(
+  slug: string | string[]
+): [string, number] {
+  if (typeof slug === undefined || slug?.length === 0) {
```

Comment:
You're absolutely right. It's a bit unnatural and verbose - and broken. It would need to be `typeof slug === 'undefined'`. This was a complete brain fart on my part caused by checking `typeof slug === 'string'` below.

To the question about when it's undefined, yeah, it's for the homepage, I believe. I didn't initially have this check and then caught it in manual testing when it came through as undefined for the homepage, IIRC.